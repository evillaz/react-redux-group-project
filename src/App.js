import React, { useEffect } from 'react';
import { Routes, Route } from 'react-router-dom';
import { useDispatch } from 'react-redux';
import Layout from './components/Layout';
import Rockets from './routes/Rockets';
import Missions from './routes/Missions';
import Profile from './routes/Profile';
import './styles/style.css';
import { rocketItems } from './Redux/RocketsSlice/RocketSlice';

function App() {
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(rocketItems());
  }, [dispatch]);

  return (
    <>
      <Routes>
        <Route path="/" element={<Layout />}>
          <Route path="profile" element={<Profile />} />
          <Route path="missions" element={<Missions />} />
          <Route path="/" element={<Rockets />} />
        </Route>
      </Routes>
    </>
  );
}

export default App;
