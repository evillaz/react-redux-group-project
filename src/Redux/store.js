import { configureStore } from '@reduxjs/toolkit';
import rocketReducer from './RocketsSlice/RocketSlice';
import missionsReducer from './missions/missionsSlice';

const store = configureStore({
  reducer: {
    rocket: rocketReducer,
    missions: missionsReducer,
  },
});

export default store;
