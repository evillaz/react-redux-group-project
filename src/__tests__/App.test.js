import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../Redux/store';
import App from '../App';

describe('App', () => {
  it('Testing of App.js', () => {
    const app = render(
      <Provider store={store}>
        <Router>
          <App />
        </Router>
      </Provider>,
    );
    expect(app).toMatchSnapshot();
  });
});
