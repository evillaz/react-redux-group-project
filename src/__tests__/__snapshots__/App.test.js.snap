// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['App Testing of App.js 1'] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="header-bar"
      >
        <a
          aria-current="page"
          class="active"
          href="/"
          style="color: rgb(0, 0, 0);"
        >
          <div
            class="logo"
          >
            <img
              alt="Space Logo"
              src="planet.png"
            />
            <h1>
              Space Travelers Hub
            </h1>
          </div>
        </a>
        <nav
          class="nav-items"
        >
          <ul>
            <li
              class="list-item"
            >
              <a
                aria-current="page"
                class="active"
                href="/"
                style="color: rgb(0, 0, 0);"
              >
                Rockets
              </a>
            </li>
            <li
              class="list-item"
            >
              <a
                class=""
                href="/missions"
                style="color: rgb(0, 0, 0);"
              >
                Missions
              </a>
            </li>
            <li
              class="list-item"
            >
              <a
                class=""
                href="/profile"
                style="color: rgb(0, 0, 0);"
              >
                My Profile
              </a>
            </li>
          </ul>
        </nav>
      </div>
      <div
        class="wrapper"
      >
        <div
          class="rocket-container"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="header-bar"
    >
      <a
        aria-current="page"
        class="active"
        href="/"
        style="color: rgb(0, 0, 0);"
      >
        <div
          class="logo"
        >
          <img
            alt="Space Logo"
            src="planet.png"
          />
          <h1>
            Space Travelers Hub
          </h1>
        </div>
      </a>
      <nav
        class="nav-items"
      >
        <ul>
          <li
            class="list-item"
          >
            <a
              aria-current="page"
              class="active"
              href="/"
              style="color: rgb(0, 0, 0);"
            >
              Rockets
            </a>
          </li>
          <li
            class="list-item"
          >
            <a
              class=""
              href="/missions"
              style="color: rgb(0, 0, 0);"
            >
              Missions
            </a>
          </li>
          <li
            class="list-item"
          >
            <a
              class=""
              href="/profile"
              style="color: rgb(0, 0, 0);"
            >
              My Profile
            </a>
          </li>
        </ul>
      </nav>
    </div>
    <div
      class="wrapper"
    >
      <div
        class="rocket-container"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
